backend/src/nodes/image_adj_nodes.py

```diff
@@ -253,3 +253,44 @@ def run(
         )
 
         return result.astype("float32") / 255
+
+
+@NodeFactory.register("chainner:image:opacity")
+class OpacityNode(NodeBase):
+    """Alpha adjustment node"""
+
+    def __init__(self):
+        """Constructor"""
+        super().__init__()
+        self.description = "Adjusts the opacity of an image."
+        self.inputs = [
+            ImageInput(),
+            SliderInput(
+                "Opacity",
+                maximum=100,
+                default=100,
+                step=0.1,
+                controls_step=1,
+                unit="%",
+            ),
+        ]
+        self.outputs = [ImageOutput()]
+        self.category = IMAGE_ADJUSTMENT
+        self.name = "Opacity"
+        self.icon = "MdOutlineOpacity"
+        self.sub = "Adjustments"
+
+    def run(self, img: np.ndarray, opacity: float) -> np.ndarray:
+        """Apply opacity adjustment to alpha channel"""
+
+        if opacity == 100:
+            return img
```

Comment:
I think this is bad.

The return value of a node should be consistent. This `if` also makes it impossible to statically type the node correctly. I'd have to type the number of output channels as `4 | InputImage.channels` which is just not as useful of just `4`.

Reply:
Please make the condition `opacity == 100 and c == 4`.

Reply:
Hmm. Was trying to avoid unnecessary conversions, but I also haven't looked into your typing system.